src/Store/FileExplorerStore.re

```diff
@@ -6,102 +6,154 @@
 open Oni_Core;
 open Oni_Model;
 
-let start = () => {
-  let (stream, _) = Isolinear.Stream.create();
-
-  let getExplorerFilesEffect = (cwd, languageInfo, iconTheme, ignored) => {
-    Isolinear.Effect.createWithDispatch(~name="explorer.open", dispatch => {
+module Effects = {
+  let load = (directory, languageInfo, iconTheme, configuration) => {
+    Isolinear.Effect.createWithDispatch(~name="explorer.load", dispatch => {
+      let ignored =
+        Configuration.getValue(c => c.filesExclude, configuration);
       let newTree =
-        FileExplorer.getDirectoryTree(cwd, languageInfo, iconTheme, ignored);
-      dispatch(Actions.SetExplorerTree(newTree));
+        FileExplorer.getDirectoryTree(
+          directory,
+          languageInfo,
+          iconTheme,
+          ignored,
+        );
+
+      dispatch(Actions.FileExplorer(TreeUpdated(newTree)));
```

Comment:
I might moderate my strong stance on making all actions in effects explicit. In cases like this were the action is internal it seems to make sense that it's implicit. Might still be useful to have an optional `onComplete` action though. I'm sure we'll find out eventually!